marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/utils/Util.java

```diff
@@ -0,0 +1,43 @@
+package com.marklogic.semantics.rdf4j.utils;
+
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+
+public class Util {
+    private static Util util = null;
+    private Util(){
+
+    }
+
+    public static Util getInstance()
+    {
+        if(util == null){
+            util = new Util();
+        }
+        return util;
+    }
+
+    /**
+     *
+     * @param s
+     * @return
+     */
+    public Value skolemize(Value s) {
+        if (s instanceof org.eclipse.rdf4j.model.BNode) {
+            return SimpleValueFactory.getInstance().createIRI("http://marklogic.com/semantics/blank/" + s.toString());
+        } else {
+            return s;
+        }
+    }
```

Comment:
At least for RDF4J the created IRI will always be valid because RDF4J just checks if the IRI string contains a colon ":"

You can find a little discussion about valid IRIs here https://github.com/eclipse/rdf4j/issues/712